### ISSUE TITLE
No images found

### ISSUE THREAD

**Issue by nickstinger** [PactInteractive/image-downloader#17]
This tool is no longer able to find any images on the page.  This started happening with a recent Chrome update.  Is it possible to ask for a fix?

Thanks.

**Comment by vdsabev** [comment]
Hi, Nick, thanks for reporting an issue!
It would be really helpful if you provided:
1. The link to the page where you're trying to get the images
2. The version of your browser, found in `chrome://settings/help`, or by clicking on the menu -> Help -> About Google Chrome
3. The version of your operating system - Windows, Linux, macOS, etc.

**Comment by Geneus** [comment]
I can confirm @nickstinger's findings. Extension was working as advertised until the most recent Chrome update. Updated last night; have issues this morning. 

1. Any page. Try these: 

- https://500px.com/photo/137554055/fresh-beef-burgers-with-cheese-vegetables-pickles-and-spicy-tomato-sauce-on-paper-over-rustic-wood-by-anna-ivanova
- https://creativemarket.com/Craftwork/421173-Singleton-UI-Pack
- https://creativemarket.com/free-goods

Oddly, it works on Twitter. Tested here: [@NASA](https://twitter.com/NASA)

2. Chrome - Version 57.0.2987.110 64-bit (Latest)
3. OS - Windows 10 Pro 64-Bit (Latest)

**Comment by vdsabev** [comment]
@Geneus thanks for the detailed info. I made some changes in #18 that improved the loading speed of the extension (and also threw in some quick features). Here are the results for me:

- https://500px.com/photo/137554055/fresh-beef-burgers-with-cheese-vegetables-pickles-and-spicy-tomato-sauce-on-paper-over-rustic-wood-by-anna-ivanova
  - **Before:** 5-6 seconds until the images are loaded
  - **After:** < 1 second
- https://creativemarket.com/Craftwork/421173-Singleton-UI-Pack
  - **Before:** I didn't wait long enough for the images to load, probably > 10 seconds (in any case -
 unacceptable)
  - **After:** 5-6 seconds until the images are loaded
- https://creativemarket.com/free-goods - about the same as the previous link

Let me know if there's any change for you after updating to 2.3.0.

There's definitely something going on. In the short term, I'll investigate and try to improve the speed more. In the long term, starting mid-April, I'll be available to work on my personal projects full-time, including this one. Expect more fixes and new features soon 😃

**Comment by adriendeotto** [comment]
@vdsabev it looks like the extension does not grab background images anymore.

For example, it does not find the header image in this page: https://www.onefootball.com/en/news/brazil-attracting-attention-of-messi-neymar-en-13481261?variable=2017-03-28T04:35:10Z

I'm using Google Chrome Version 57.0.2987.110 (64-bit).

**Comment by vdsabev** [comment]
@adriendeotto you're right, it was a late night and I missed something (that now looks obvious). I made some changes in #19 and just published a fix. Unfortunately, performance is still subpar 😞

Let me know how 2.3.1 works out for you! Here's a shot:

![onefootball](https://cloud.githubusercontent.com/assets/1333274/24397916/58d032ac-13b0-11e7-8082-03d8c33a1238.png)

**Comment by adriendeotto** [comment]
@vdsabev amazing: it's now working perfectly again! Thank you very much :)

**Comment by nickstinger** [comment]
At the time I reported this issue, I couldn't download images from any site.  Now the extension seems to be working for most sites, except the one site I am particularly concerned with.  The site is not public, but the image is easily selected by using ```document.getElementsByTagName("img")```. I disabled all filters in the menu but for some reason this image never gets selected. I'll see if I can't get some more helpful information.

**Comment by nickstinger** [comment]
![image](https://cloud.githubusercontent.com/assets/4709341/24583805/2819473a-1792-11e7-8db5-53096827b7e1.png)

![image](https://cloud.githubusercontent.com/assets/4709341/24583806/2dcc0b7c-1792-11e7-995d-a5b2af9960b6.png)

**Comment by EC-O-DE** [comment]
Check this ext called "Imagus" - the Sieve file is pretty good at getting pics from almost any site - even forging a higher quality version if available!

**Comment by vdsabev** [comment]
Closing this issue as it hasn't had activity in a while. If the issue persists, please leave a comment or open a new issue. Thanks!
